python_client/src/experiment.py

```diff
@@ -0,0 +1,101 @@
+from typing import List
+import requests
+
+from .scan import Scan
+from .exception import MIQAAPIError
+
+
+class Experiment:
+    """
+    Attributes:
+      id, name, scans, project
+    Functions:
+      get_scan_by_id, add_scan,
+      print_all_objects
+
+    """
+
+    def __init__(
+        self,
+        id: str,
+        name: str,
+        project,
+        note: str,
+        scans: List[dict] = [],
+        **kwargs,
+    ):
+        self.id = id
+        self.name = name
+        self.scans = [Scan(**dict(scan, experiment=self)) for scan in scans]
+        self.project = project
+        self.note = note
+
+    def get_scan_by_id(self, id: str):
+        if self.scans:
+            matches = [scan for scan in self.scans if scan.id == id]
+            if len(matches) == 1:
+                return matches[0]
+        api_path = f"scans/{id}"
+        response = requests.get(
+            f"{self.project.MIQA.url}/{api_path}",
+            headers=self.project.MIQA.headers,
+        )
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError:
```

Comment:
Rather than `try / except` you can check `response.ok`

Also check the status code in this specific case, since we only want to return `None` if it's HTTP 404.

Reply:
Changed in [bd731b1](https://github.com/OpenImaging/miqa/pull/533/commits/bd731b13a808d215e55f72dcf4911101d5c6d6d6)